llvm/lib/Transforms/Coroutines/CoroFrame.cpp

```diff
@@ -1550,13 +1556,16 @@ struct AllocaUseVisitor : PtrUseVisitor<AllocaUseVisitor> {
   }
 
   void visitIntrinsicInst(IntrinsicInst &II) {
+    if (II.getIntrinsicID() == Intrinsic::lifetime_end)
```

Comment:
Should this also take into account if the lifetime marker only applies to a subrange of the alloca (like below)?

Reply:
Done (but there's no logic change anyways since we don't check `LifetimeEndBBs` if there are no `LifetimeStarts`).

Reply:
There could be a "full" `lifetime.start` followed by a "partial" `lifetime.end`, so I think this matters. But the latest version of the PR handles that nicely.